### PR TITLE
fix: FixedBitString might work as expected

### LINKED ISSUE #383: `rasn::uper::decode` fails
Hi, thanks for the great work so far. 

I am working with SPATEM and MAPEM message types from [ETSI TS 103 301](https://forge.etsi.org/rep/ITS/asn1/is_ts103301). Using the rasn compiler, I successfully generated Rust code from the provided ASN.1 schemas.

However, I am encountering issues when decoding these messages:

- The MapData structure within MAPEM fails to decode.
- The SpatData structure within SPATEM also fails to decode.

The ItsPduHeader for both message types decodes successfully.

<details>
  <summary>Mapem wireshark</summary>
Bildschirmfoto 2024-11-27 um 18 43 32
</details>

<details>
  <summary>Mapem decoding test</summary>

```rust
use v2x_messages_rs::dsrc::*;
use v2x_messages_rs::its__container::ItsPduHeader;
use v2x_messages_rs::mapem__pdu__descriptions::MAPEM;

#[test]
fn derive_mapem() {
    const MAPEM: [u8; 1121] = [
        0x2, 0x5, 0x0, 0xff, 0xe0, 0xe9, 0x8, 0x0, 0x3, 0x83, 0xb0, 0x6e, 0xc1, 0x83, 0x1e, 0xcc,
        0x1a, 0xc0, 0xc, 0x46, 0xd6, 0x40, 0x8a, 0x9e, 0xb, 0xad, 0x8e, 0x67, 0x7e, 0xf9, 0x80,
        0x4b, 0x0, 0xa2, 0xb6, 0x2a, 0x24, 0x1, 0x14, 0x0, 0x0, 0x0, 0x8, 0x6e, 0x83, 0x26, 0x1,
        0xe, 0x2e, 0x66, 0xc4, 0x6, 0x6b, 0xcc, 0xa3, 0x85, 0x8d, 0x24, 0x0, 0x52, 0x58, 0xd1, 0x0,
        0x0, 0x80, 0xac, 0x6c, 0x80, 0x0, 0x40, 0x96, 0x38, 0x40, 0x0, 0x20, 0x64, 0x80, 0x42,
        0x80, 0x0, 0x0, 0x0, 0xd, 0xcd, 0x5, 0xf9, 0x29, 0x60, 0xfb, 0x11, 0xe8, 0x2c, 0x62, 0x0,
        0x0, 0x41, 0x9, 0x0, 0xc5, 0x0, 0x0, 0x0, 0x1, 0x1b, 0x93, 0x4e, 0x6e, 0xc2, 0x2d, 0x57,
        0xf2, 0x40, 0x5, 0x2a, 0x1a, 0xc3, 0x52, 0x60, 0x2c, 0x66, 0x0, 0x0, 0x41, 0x49, 0x1, 0x5,
        0x0, 0x0, 0x0, 0x1, 0x13, 0x24, 0x83, 0xb1, 0xc, 0x1, 0x63, 0xa5, 0x22, 0xe4, 0xe5, 0xa0,
        0x2, 0x9a, 0xc5, 0x90, 0x0, 0x4, 0x19, 0x62, 0xa8, 0x0, 0x2, 0xe, 0xb1, 0x44, 0x0, 0x1,
        0xf, 0x58, 0x98, 0x80, 0x0, 0x93, 0x92, 0x2, 0x92, 0x0, 0x0, 0x0, 0x2, 0x2c, 0xf4, 0xf7,
        0xd1, 0x43, 0x8e, 0x7e, 0x23, 0x2b, 0xf1, 0x10, 0x1, 0x49, 0x62, 0x64, 0x0, 0x4, 0x10,
        0xb1, 0xa8, 0x0, 0x2, 0x9, 0x58, 0x91, 0x0, 0x1, 0x7, 0x12, 0x3, 0x12, 0x0, 0x0, 0x0, 0x2,
        0x2e, 0xe, 0xfe, 0x65, 0x44, 0x1f, 0x2c, 0x20, 0x2, 0x96, 0xe, 0xfa, 0x99, 0xa8, 0x16,
        0x37, 0x0, 0x0, 0x41, 0x44, 0x80, 0xe4, 0x80, 0x0, 0x0, 0x0, 0xe, 0xe7, 0x58, 0x53, 0x2c,
        0x21, 0xcd, 0x61, 0xd0, 0x2c, 0x72, 0x0, 0x0, 0x82, 0xc9, 0x2, 0x9, 0x0, 0x0, 0x0, 0x1,
        0x1e, 0x1a, 0xb1, 0x80, 0x38, 0x94, 0xf3, 0x95, 0x71, 0x5, 0xca, 0x52, 0x0, 0x28, 0x2c,
        0x61, 0x0, 0x0, 0x83, 0x9, 0x2, 0x49, 0x0, 0x0, 0x0, 0x4, 0x1e, 0x5e, 0xf2, 0x12, 0x38,
        0x93, 0xf4, 0x8, 0x30, 0xff, 0x8d, 0x85, 0xe2, 0xf4, 0x84, 0xbf, 0x3d, 0x0, 0x80, 0xa,
        0x48, 0x80, 0x75, 0x8b, 0x15, 0x8c, 0xa0, 0x0, 0x10, 0x6a, 0xc5, 0x84, 0x0, 0x8, 0xa0,
        0x90, 0x28, 0xd0, 0x0, 0x0, 0x0, 0x32, 0x6a, 0x11, 0x34, 0x89, 0x3b, 0x1e, 0xa5, 0xc4,
        0x1f, 0xb3, 0xb4, 0x1c, 0xa0, 0xee, 0x93, 0xa6, 0x62, 0x52, 0x0, 0x29, 0x2c, 0x50, 0x80,
        0x0, 0xc4, 0x16, 0x2a, 0x40, 0x0, 0x62, 0x2b, 0x16, 0x20, 0x0, 0x31, 0x22, 0x40, 0xb3,
        0x40, 0x0, 0x0, 0x0, 0x9, 0xa7, 0x8c, 0xa6, 0x65, 0xb9, 0x10, 0x94, 0x2e, 0x5, 0x89, 0x40,
        0x0, 0x18, 0x99, 0x20, 0x61, 0xa0, 0x0, 0x0, 0x0, 0x44, 0xd3, 0x46, 0x3f, 0x55, 0xfb, 0xfa,
        0xa1, 0x28, 0x0, 0xac, 0xcc, 0xde, 0x6e, 0x14, 0x0, 0x52, 0x6e, 0xe9, 0x41, 0x38, 0x2c,
        0x4e, 0x0, 0x0, 0xc5, 0x9, 0x3, 0x4d, 0x0, 0x0, 0x0, 0x2, 0x26, 0x97, 0xf1, 0x52, 0x8f,
        0x6f, 0xd4, 0x7, 0x7, 0xab, 0x94, 0xa3, 0xbd, 0xca, 0x10, 0x1, 0x4d, 0x63, 0x88, 0x0, 0x6,
        0x2a, 0xb1, 0xb4, 0x0, 0x3, 0x16, 0x58, 0xd2, 0x0, 0x1, 0x8b, 0xac, 0x64, 0x40, 0x0, 0xca,
        0x49, 0x3, 0x91, 0x0, 0x0, 0x0, 0x1, 0x1e, 0x1c, 0x83, 0xc8, 0x37, 0x62, 0x8, 0x85, 0x70,
        0x6e, 0x3f, 0x5a, 0x0, 0x29, 0x2c, 0x49, 0x0, 0x1, 0x7, 0x56, 0x26, 0x80, 0x0, 0x83, 0xcb,
        0x1c, 0x10, 0x0, 0x42, 0xa2, 0x40, 0xf4, 0x40, 0x0, 0x0, 0x0, 0xc7, 0x99, 0x11, 0x28, 0x91,
        0xe4, 0x33, 0x2d, 0xd2, 0xf4, 0x47, 0x42, 0x0, 0x2a, 0x9f, 0xa0, 0xab, 0x10, 0x1, 0x4a,
        0xf2, 0xba, 0x81, 0xb8, 0xb, 0x16, 0x80, 0x0, 0x41, 0xc2, 0x41, 0x4, 0x40, 0x0, 0x0, 0x0,
        0x47, 0xac, 0xf1, 0x86, 0x11, 0xe3, 0xa2, 0xfd, 0x56, 0xe8, 0x33, 0xcd, 0x20, 0x2, 0x82,
        0xc5, 0x60, 0x0, 0x10, 0x6c, 0x90, 0x45, 0x10, 0x0, 0x0, 0x0, 0x21, 0xf0, 0x7c, 0x79, 0x44,
        0x79, 0x2c, 0xd2, 0x1, 0xb8, 0xe4, 0x96, 0x5, 0x79, 0xad, 0x41, 0xca, 0x25, 0x8c, 0x10,
        0x0, 0x20, 0xc2, 0xc6, 0x48, 0x0, 0x10, 0x65, 0x62, 0x90, 0x0, 0x8, 0x34, 0x20, 0x24, 0x24,
        0x0, 0x0, 0x0, 0x0, 0x54, 0xf1, 0xfb, 0x19, 0xd2, 0x99, 0xe6, 0x20, 0x26, 0x24, 0x0, 0x0,
        0x0, 0x0, 0x9a, 0x68, 0xfa, 0xc6, 0x74, 0x96, 0x7b, 0x88, 0xa, 0x11, 0x0, 0x0, 0x0, 0x0,
        0x3f, 0x19, 0x68, 0x88, 0x51, 0x6b, 0xfb, 0x2, 0x2, 0xa4, 0x40, 0x0, 0x0, 0x0, 0xf, 0x9e,
        0x19, 0xb0, 0x14, 0x5d, 0x7f, 0xc0, 0x80, 0xb1, 0x10, 0x0, 0x0, 0x0, 0x3, 0xde, 0x26, 0x4d,
        0x85, 0x17, 0xdf, 0xde, 0x20, 0x30, 0x64, 0x0, 0x0, 0x0, 0x1, 0x33, 0xca, 0x74, 0x28, 0x5b,
        0xe5, 0x3e, 0x62, 0x3, 0x26, 0x40, 0x0, 0x0, 0x0, 0x13, 0x3c, 0xa7, 0xb9, 0xc5, 0xc3, 0xf4,
        0x30, 0x20, 0x34, 0x84, 0x0, 0x0, 0x0, 0x0, 0xe7, 0xe0, 0xd, 0x45, 0xbd, 0x28, 0xfa, 0xe8,
        0x0, 0x82, 0x3, 0x68, 0x40, 0x0, 0x0, 0x0, 0xe, 0xa4, 0xa1, 0x77, 0x1b, 0xcf, 0x4f, 0x92,
        0x20, 0x38, 0x84, 0x0, 0x0, 0x0, 0x0, 0xec, 0x90, 0x1d, 0x71, 0xbd, 0x2c, 0xf8, 0x86, 0x83,
        0xa2, 0x2c, 0x0, 0x10, 0x0, 0x0, 0x5, 0xc2, 0x20, 0x8e, 0x60, 0x9f, 0xa0, 0x20, 0x4, 0xe4,
        0x16, 0x3d, 0x0, 0x1, 0x23, 0xe6, 0x83, 0xc2, 0x2c, 0x0, 0x10, 0x0, 0x0, 0x5, 0x43, 0xd0,
        0xe7, 0x82, 0x7e, 0x88, 0x79, 0x6f, 0x4, 0x0, 0x40, 0x58, 0xec, 0x0, 0x4, 0x95, 0x9a, 0xf,
        0x88, 0xb0, 0x0, 0x40, 0x0, 0x0, 0x15, 0x9, 0x7, 0x34, 0x9, 0xfa, 0x22, 0x16, 0x49, 0x90,
        0x1, 0x1, 0x64, 0x10, 0x0, 0x14, 0x40, 0x68, 0x40, 0x22, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x44,
        0x3e, 0xe0, 0x82, 0x7e, 0x80, 0x71, 0x6a, 0xc0, 0x58, 0xfc, 0x0, 0x5, 0x16, 0x1a, 0x10,
        0x91, 0x30, 0x0, 0x40, 0x0, 0x0, 0x15, 0x82, 0x19, 0x94, 0x9, 0xfa, 0x1, 0x34, 0x84, 0x81,
        0x64, 0x50, 0x0, 0x16, 0x42, 0x68, 0x44, 0x44, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5e, 0x75, 0x6d,
        0xce, 0x9, 0xfa, 0x22, 0x83, 0x86, 0xd0, 0x1, 0x1, 0x64, 0x30, 0x0, 0x16, 0x5a, 0x68, 0x46,
        0x44, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5e, 0xbe, 0x2e, 0x60, 0x9, 0xfa, 0x21, 0x61, 0x74, 0xd0,
        0x1, 0x1, 0x64, 0x90, 0x0, 0x18, 0x44, 0x68, 0x48, 0x44, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5f,
        0xd9, 0xf1, 0x6c, 0x9, 0xfa, 0x2, 0xbe, 0x91, 0x1, 0x64, 0x70, 0x0, 0x18, 0x5c, 0x68, 0x4a,
        0x66, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x66, 0x55, 0x93, 0x4a, 0x82, 0x7e, 0x80, 0x82, 0x6c, 0x70,
        0x59, 0x34, 0x0, 0x6, 0x91, 0x9a, 0x13, 0x19, 0xb0, 0x0, 0x40, 0x0, 0x0, 0x19, 0x91, 0x64,
        0x1f, 0x20, 0x9f, 0xa2, 0x1f, 0xc3, 0xf9, 0x0, 0x10, 0x16, 0x4b, 0x0, 0x1, 0xa5, 0xe6,
        0x84, 0xe6, 0x6c, 0x0, 0x10, 0x0, 0x0, 0x6, 0x64, 0x3e, 0xf9, 0x78, 0x27, 0xe8, 0x88, 0x26,
        0xc7, 0x40, 0x4, 0x5, 0x94, 0x40, 0x0, 0x71, 0x21, 0xa1, 0x41, 0x9b, 0x0, 0x4, 0x0, 0x0,
        0x1, 0x98, 0xfb, 0x35, 0xc8, 0x9, 0xfa, 0x2, 0x31, 0x5b, 0xc1, 0x64, 0xf0, 0x0, 0x1c, 0x60,
        0x68, 0x52, 0x88, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5f, 0x45, 0xd, 0x5e, 0x9, 0xfa, 0x2, 0x95,
        0x7d, 0xc1, 0x65, 0x50, 0x0, 0x1e, 0x4a, 0x68, 0x54, 0x88, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5e,
        0x3, 0x88, 0x66, 0x9, 0xfa, 0x21, 0x86, 0x74, 0xd0, 0x1, 0x1, 0x65, 0x30, 0x0, 0x1e, 0x62,
        0x68, 0x56, 0x88, 0xc0, 0x1, 0x0, 0x0, 0x0, 0x5d, 0xc7, 0x87, 0xcc, 0x9, 0xfa, 0x22, 0x76,
        0x87, 0xd0, 0x1, 0x1, 0x65, 0x90, 0x0, 0x20, 0x4c, 0x68, 0x58, 0x88, 0xc0, 0x1, 0x0, 0x0,
        0x0, 0x5c, 0xdd, 0x84, 0x3e, 0x9, 0xfa, 0x1, 0x78, 0x76, 0x1, 0x65, 0x70, 0x0, 0x20, 0x64,
    ];

    const ITS_PDU_HEADER: [u8; 6] = [0x2, 0x5, 0x0, 0xff, 0xe0, 0xe9];
    let its_pdu_header = rasn::uper::decode::<ItsPduHeader>(&ITS_PDU_HEADER).unwrap();
    println!("ITS_PDU_HEADER: {:?}\n", its_pdu_header);

    let mapem = rasn::uper::decode::<MAPEM>(&MAPEM).unwrap();
    println!("MAPEM: {:?}\n", mapem);
}
```

Error: 

```bash
thread 'derive_mapem' panicked at tests/mapem.rs:155:53:
called `Result::unwrap()` on an `Err` value: DecodeError { kind: FieldError { name: "MAPEM.map", nested: DecodeError { kind: FieldError { name: "MapData.intersections", nested: DecodeError { kind: FieldError { name: "IntersectionGeometry.speed_limits", nested: DecodeError { kind: FieldError { name: "RegulatorySpeedLimit.r_type", nested: DecodeError { kind: EnumerationIndexNotFound { index: 49, extended_list: true }, codec: Uper } }, codec: Uper } }, codec: Uper } }, codec: Uper } }, codec: Uper }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test derive_mapem ... FAILED
```

</details>

<details>
  <summary>Spatem decoding test</summary>

```rust
use v2x_messages_rs::dsrc::SPAT;
use v2x_messages_rs::its__container::ItsPduHeader;

#[test]
fn derive_spatem() {
    const SPAT_DATA: [u8; 710] = [
        0x47, 0x23, 0x32, 0x01, 0x88, 0x9a, 0x10, 0xdb, 0xc8, 0x10, 0x20, 0x07, 0x23, 0x32, 0x42,
        0xc0, 0x0f, 0x20, 0x13, 0x43, 0x70, 0x31, 0x10, 0x31, 0x60, 0x31, 0x17, 0xa3, 0x38, 0x1b,
        0x58, 0x1b, 0x80, 0x1b, 0x5b, 0xd0, 0xdc, 0x10, 0xf4, 0x11, 0x08, 0x10, 0xf5, 0xe8, 0xce,
        0x09, 0x2e, 0x09, 0x38, 0x09, 0x2e, 0xf9, 0x00, 0x0c, 0x00, 0x40, 0x01, 0x40, 0x07, 0x00,
        0x20, 0x00, 0x90, 0x02, 0x80, 0x0b, 0x00, 0x30, 0x00, 0xd2, 0x02, 0x34, 0x67, 0x02, 0xc6,
        0x02, 0xc6, 0x02, 0xc6, 0x72, 0x1b, 0x81, 0xd6, 0x01, 0xd6, 0x01, 0xd6, 0x3d, 0x19, 0xc0,
        0xfc, 0x80, 0xfc, 0x80, 0xfc, 0x9e, 0x86, 0xe0, 0x9b, 0x00, 0x9b, 0x00, 0x9b, 0x0f, 0xa0,
        0x04, 0x00, 0x11, 0x00, 0x48, 0x04, 0x50, 0x11, 0x80, 0x47, 0x00, 0x4c, 0x01, 0x40, 0x05,
        0x40, 0x16, 0x00, 0x5c, 0x80, 0xcd, 0x0d, 0xc0, 0xc3, 0x00, 0xc5, 0x80, 0xc4, 0x5c, 0x8c,
        0xe0, 0x6a, 0x40, 0x6b, 0x80, 0x6a, 0xee, 0x43, 0x70, 0x43, 0x80, 0x44, 0x20, 0x43, 0xd7,
        0xa3, 0x38, 0x23, 0xf0, 0x24, 0x68, 0x24, 0x1b, 0xf0, 0x01, 0x80, 0x06, 0x40, 0x1d, 0x00,
        0x78, 0x01, 0xb0, 0x07, 0x00, 0x1f, 0x00, 0x80, 0x02, 0x10, 0x08, 0x80, 0x23, 0x00, 0x90,
        0x01, 0xa2, 0x04, 0x34, 0x67, 0x02, 0xcb, 0x02, 0xcb, 0x02, 0xcb, 0x6a, 0x1b, 0x81, 0xd6,
        0x01, 0xd6, 0x01, 0xd6, 0x3d, 0x19, 0xc0, 0xfc, 0x80, 0xfd, 0xc0, 0xfd, 0xde, 0x86, 0xe0,
        0x9a, 0x60, 0x9b, 0x00, 0x9b, 0x0f, 0xe0, 0x09, 0x40, 0x26, 0x00, 0x9c, 0x02, 0x80, 0x0a,
        0x40, 0x2a, 0x00, 0xac, 0x02, 0xc0, 0x14, 0x80, 0x53, 0x00, 0xc0, 0x03, 0x10, 0x0c, 0x80,
        0x2d, 0x00, 0xb8, 0x82, 0x4d, 0x19, 0xc0, 0xb4, 0x00, 0xb4, 0x00, 0xb4, 0x1e, 0x86, 0xe0,
        0x70, 0x80, 0x70, 0x80, 0x70, 0x8f, 0x46, 0x70, 0x3f, 0xc0, 0x3f, 0xc0, 0x3f, 0xc7, 0xa1,
        0xb8, 0x25, 0x80, 0x25, 0x80, 0x25, 0x83, 0xc4, 0x03, 0x30, 0x12, 0x08, 0x28, 0xd1, 0x9c,
        0x0a, 0xf0, 0x0a, 0xf0, 0x0a, 0xf1, 0xe8, 0x6e, 0x07, 0x26, 0x07, 0x26, 0x07, 0x26, 0xf4,
        0x67, 0x03, 0xe8, 0x03, 0xe8, 0x03, 0xe8, 0x7a, 0x1b, 0x82, 0x5f, 0x82, 0x5f, 0x82, 0x5f,
        0xbc, 0x40, 0x34, 0x01, 0x24, 0x82, 0xcd, 0x0d, 0xc0, 0xbb, 0x80, 0xbe, 0x00, 0xbc, 0xda,
        0x8c, 0xe0, 0x6f, 0xe0, 0x71, 0x20, 0x70, 0x8e, 0x43, 0x70, 0x41, 0xa0, 0x42, 0x40, 0x41,
        0xf7, 0xa3, 0x38, 0x25, 0x58, 0x25, 0xa8, 0x25, 0x83, 0xc4, 0x03, 0x50, 0x12, 0x88, 0x30,
        0xd0, 0xdc, 0x0c, 0x08, 0x0c, 0x30, 0x0c, 0x1d, 0xc8, 0xce, 0x06, 0xa4, 0x06, 0xb8, 0x06,
        0xae, 0xe4, 0x37, 0x04, 0x2e, 0x04, 0x38, 0x04, 0x33, 0x7a, 0x33, 0x82, 0x3f, 0x02, 0x44,
        0x02, 0x41, 0xbc, 0x40, 0x36, 0x01, 0x2c, 0x83, 0x4d, 0x19, 0xc0, 0xb2, 0xc0, 0xb2, 0xc0,
        0xb2, 0xdc, 0x86, 0xe0, 0x72, 0x60, 0x72, 0x60, 0x72, 0x6f, 0x46, 0x70, 0x3f, 0x70, 0x3f,
        0x70, 0x3f, 0x77, 0xa1, 0xb8, 0x25, 0xf8, 0x25, 0xf8, 0x25, 0xfb, 0xc4, 0x03, 0x70, 0x13,
        0x08, 0x38, 0xd1, 0x9c, 0x0b, 0x18, 0x0b, 0x18, 0x0b, 0x19, 0xc8, 0x6e, 0x07, 0x44, 0x07,
        0x44, 0x07, 0x44, 0xf4, 0x67, 0x03, 0xf2, 0x03, 0xf2, 0x03, 0xf2, 0x7a, 0x1b, 0x82, 0x67,
        0x02, 0x67, 0x02, 0x67, 0x3c, 0x40, 0x38, 0x01, 0x34, 0x83, 0xcd, 0x0d, 0xc0, 0xbc, 0xc0,
        0xbc, 0xc0, 0xbc, 0xdc, 0x8c, 0xe0, 0x6e, 0xa0, 0x6f, 0x40, 0x6f, 0x4f, 0x43, 0x70, 0x41,
        0xa0, 0x42, 0x40, 0x41, 0xf7, 0xa3, 0x38, 0x25, 0x08, 0x25, 0x58, 0x25, 0x33, 0xc4, 0x03,
        0x90, 0x13, 0x88, 0x40, 0xd0, 0xdc, 0x0b, 0xf4, 0x0c, 0x1c, 0x0c, 0x09, 0xa8, 0xce, 0x06,
        0x86, 0x06, 0xa4, 0x06, 0x90, 0xe4, 0x37, 0x04, 0x29, 0x04, 0x38, 0x04, 0x2e, 0x7a, 0x33,
        0x82, 0x37, 0x82, 0x3f, 0x02, 0x3a, 0x3c, 0x40, 0x3a, 0x01, 0x3c, 0x86, 0x4d, 0x0d, 0xc0,
        0xbc, 0xc0, 0xbe, 0x00, 0xbc, 0xdc, 0x8c, 0xe0, 0x68, 0x60, 0x69, 0x00, 0x68, 0x6f, 0x43,
        0x70, 0x41, 0xf0, 0x42, 0x40, 0x41, 0xf7, 0xa3, 0x38, 0x23, 0x78, 0x23, 0xa0, 0x23, 0x7b,
        0xcc, 0x03, 0xc0, 0x0f, 0x40, 0x3e, 0x00, 0xfc, 0x87, 0x0d, 0x19, 0xc0, 0xad, 0xc0, 0xad,
        0xc0, 0xad, 0xde, 0x86, 0xe0, 0x71, 0xc0, 0x71, 0xc0, 0x71, 0xcf, 0x46, 0x70, 0x3e, 0x30,
        0x3e, 0x30, 0x3e, 0x37, 0xa1, 0xb8, 0x25, 0xd0, 0x25, 0xd0, 0x25, 0xd3, 0xc4, 0x05, 0x00,
        0x10, 0x08, 0x74, 0xd1, 0x9c, 0x0b, 0x18, 0x0b, 0x18, 0x0b, 0x19, 0xe8, 0x6e, 0x07, 0x12,
        0x07, 0x12, 0x07, 0x12, 0xf4, 0x67, 0x03, 0xf2, 0x03, 0xf2, 0x03, 0xf2, 0x7a, 0x1b, 0x82,
        0x5a, 0x82, 0x5a, 0x82, 0x5a, 0xbc, 0x40, 0x41, 0x01, 0x44, 0x86, 0x8d, 0x0d, 0xc0, 0xbc,
        0xc0, 0xbf, 0x40, 0xbe, 0x1a, 0x8c, 0xe0, 0x6d, 0x60, 0x6e, 0xa0, 0x6e, 0x0e, 0x43, 0x70,
        0x41, 0xf0, 0x42, 0x90, 0x42, 0x47, 0xa3, 0x38, 0x24, 0xb8, 0x25, 0x08, 0x24, 0xe3, 0xc8,
        0x04, 0x20, 0x10, 0xc0, 0x44,
    ];

    const ITS_PDU_HEADER: [u8; 6] = [0x02, 0x04, 0x00, 0x00, 0x00, 0x00];
    let its_pdu_header: ItsPduHeader = rasn::uper::decode::<ItsPduHeader>(&ITS_PDU_HEADER).unwrap();
    let spat = rasn::uper::decode::<SPAT>(&SPAT_DATA).unwrap();

    println!("ITS_PDU_HEADER: {:?}\n", its_pdu_header);
    println!("SPAT: {:?}\n", spat);
}
```

Error: 

```bash
running 1 test
thread 'derive_spatem' panicked at tests/spatem.rs:59:55:
called `Result::unwrap()` on an `Err` value: DecodeError { kind: FieldError { name: "SPAT.intersections", nested: DecodeError { kind: FieldError { name: "IntersectionState.status", nested: DecodeError { kind: FixedStringConversionFailed { tag: Tag { class: Universal, value: 3 }, expected: 16, actual: 2 }, codec: Uper } }, codec: Uper } }, codec: Uper }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test derive_spatem ... FAILED
```
</details>

Do you have any suggestions for debugging this issue? As someone relatively new to ASN.1, I’m feeling a bit stuck and would greatly appreciate any guidance or tips on how to proceed.

Thank you in advance for your help!

### PR DESCRIPTION
**Pull request by Nicceboy**
PartiallyFixes #383

I am not sure if the current way is the most convenient way to use the type, but without wrapping it with newtype, I will not get much better.

### CODE REVIEW

File: src/types/strings/bit.rs [hunk 1]
```diff
@@ -16,7 +16,28 @@ use crate::prelude::*;
 /// ```
 pub type BitString = bitvec::vec::BitVec<u8, bitvec::order::Msb0>;
 ///  A fixed length `BIT STRING` type.
+///
+/// IMPORTANT: While N describes the number of bits, also the internal size is for N amount of bytes.
+/// When accessing the bits, use &array[..N] to get the correct  bits, for example.
+/// Also when setting the bits, they should start from the "left". See example.
```
Comment (XAMPPRocky) [inline src/types/strings/bit.rs:22]: I think the better way to describe it is that it expects "big endian" style encoding.

**Comment by XAMPPRocky** [comment]
> I am not sure if the current way is the most convenient way to use the type, but without wrapping it with newtype, I will not get much better.

I want to clarify that, that is a possibility. If we need to wrap types to provide a better interface we can and should do that. This is totally fine for now though.

**Comment by Nicceboy** [comment]
I'll link #76 if it can be then improved at the same time.

**Comment by Nicceboy** [comment]
What if we would just use type `[bool; N]` instead? We don't lose in size since BitArray takes the same space, and `[bool; N]` would be also constant (it is not currently). It would be easier to operate and we reduce the dependency from `BitVec` crate.

**Pull request merged**